File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderDate.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.DateRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.StyledText;

import static com.ulyp.ui.util.CssClass.*;

public class RenderDate extends RenderedObject {
    public RenderDate(DateRepresentation repr, RenderSettings renderSettings) {
        super(repr.getType());

        if (renderSettings.showTypes()) {
            super.getChildren().add(StyledText.of(repr.getType().getName(), CALL_TREE_TYPE_NAME));
            super.getChildren().add(StyledText.of(": ", CALL_TREE_PLAIN_TEXT));
        }
        super.getChildren().add(StyledText.of(repr.getDatePrinted(), CALL_TREE_NUMBER));
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderDate.kt
package com.ulyp.ui.renderers

import com.ulyp.core.recorders.DateRecord
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of

class RenderDate(record: DateRecord, renderSettings: RenderSettings) : RenderedObject(record.type) {

    init {
        if (renderSettings.showTypes()) {
            super.getChildren().add(of(record.type.name, CssClass.CALL_TREE_TYPE_NAME))
            super.getChildren().add(of(": ", CssClass.CALL_TREE_NODE_SEPARATOR))
        }
        super.getChildren().add(of(record.datePrinted, CssClass.CALL_TREE_NUMBER))
    }
}